presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -162,85 +192,74 @@ public static Domain getDomain(Type type, long rowCount, Statistics<?> statistic
             throw new VerifyException("Impossible boolean statistics");
         }
 
-        if ((type.equals(BIGINT) || type.equals(TINYINT) || type.equals(SMALLINT) || type.equals(INTEGER)) && (statistics instanceof LongStatistics || statistics instanceof IntStatistics)) {
-            ParquetIntegerStatistics parquetIntegerStatistics;
-            if (statistics instanceof LongStatistics) {
-                LongStatistics longStatistics = (LongStatistics) statistics;
-                if (longStatistics.genericGetMin() > longStatistics.genericGetMax()) {
-                    failWithCorruptionException(failOnCorruptedParquetStatistics, column, id, longStatistics);
+        if ((type.equals(BIGINT) || type.equals(TINYINT) || type.equals(SMALLINT) || type.equals(INTEGER))) {
+            List<Range> ranges = new ArrayList<>();
+            for (int i = 0; i < minimums.size(); i++) {
+                long min = asLong(minimums.get(i));
+                long max = asLong(maximums.get(i));
+                if (isStatisticsOverflow(type, min, max)) {
                     return Domain.create(ValueSet.all(type), hasNullValue);
                 }
-                parquetIntegerStatistics = new ParquetIntegerStatistics(longStatistics.genericGetMin(), longStatistics.genericGetMax());
-            }
-            else {
-                IntStatistics intStatistics = (IntStatistics) statistics;
-                if (intStatistics.genericGetMin() > intStatistics.genericGetMax()) {
-                    failWithCorruptionException(failOnCorruptedParquetStatistics, column, id, intStatistics);
-                    return Domain.create(ValueSet.all(type), hasNullValue);
-                }
-                parquetIntegerStatistics = new ParquetIntegerStatistics((long) intStatistics.getMin(), (long) intStatistics.getMax());
-            }
-            if (isStatisticsOverflow(type, parquetIntegerStatistics)) {
-                return Domain.create(ValueSet.all(type), hasNullValue);
-            }
-            return createDomain(type, hasNullValue, parquetIntegerStatistics);
-        }
 
-        if (type.equals(REAL) && statistics instanceof FloatStatistics) {
-            FloatStatistics floatStatistics = (FloatStatistics) statistics;
-            if (floatStatistics.genericGetMin() > floatStatistics.genericGetMax()) {
-                failWithCorruptionException(failOnCorruptedParquetStatistics, column, id, floatStatistics);
-                return Domain.create(ValueSet.all(type), hasNullValue);
+                ranges.add(Range.range(type, min, true, max, true));
             }
 
-            if (floatStatistics.genericGetMin().isNaN() || floatStatistics.genericGetMax().isNaN()) {
-                return Domain.create(ValueSet.all(type), hasNullValue);
-            }
+            return Domain.create(ValueSet.ofRanges(ranges), hasNullValue);
```

Comment:
do we need a check here for `ranges.size() > 0`?

Reply:
sure. will add